Fix some types and cleanup in Specimen Definition Form

yash-learner opened the pull request:
## Proposed Changes

- Fix form getting not submitted because of zod validation fail in typetested fields while updating
- Cleaned up some types and also after some reading understood ? is better than null`  in types and .optional  is better than .nullable().  -> This made code to be more Readale -> which made removing explicit value set in input component

from
```
<Input
  placeholder={t("uri")}
  {...field}
  value={field.value || ""}
/> 
```

to this

```
<Input placeholder={t("uri")} {...field} />
```

- Some tiny cleanups and some tiny improvements - Like currently when changing tabs in minimum quantity it clear the value -> Now it clear only after typed in other tab
- Fix the minimum_quantity: {} being passed when fields are empty → this fixed container section being with title in detail page.
- Fixed  issues with `ComboboxQunatity` fields

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA in Mobile Devices
- [ ] Completion of QA in Desktop Devices

yash-learner committed:
Clean up and improve types

yash-learner committed:
Update the relevant files for change in type

yash-learner committed:
Merge branch 'labs-v2' into some-fixes-and-cleanup-in-sdf

yash-learner committed:
Fix the some types

yash-learner committed:
Clean up the minimun value when  it is {}

- tried fix it via RHF but did not work so cleaning it up on submit

yash-learner committed:
Merge branch 'labs-v2' into some-fixes-and-cleanup-in-sdf

abhimanyurajeesh reviewed (changes requested):
Let's have validation to make sure that the user have selected the unit also, not just values.
(selectedUnit)

yash-learner committed:
When user just enters the the value., auto-select to first unit

- This would fix user not selecting the unti and just entering the value.
- If they want they can select a different unit from popover

yash-learner commented:
> Let's have validation to make sure that the user have selected the unit also, not just values. (selectedUnit)

@abhimanyurajeesh fixed it 

I decided to preserve the old behaviour of auto-selecting the first unit, instead of getting selected without typing anything now it selects after user types some value 

Because it is better UX, and they still have option to select a different unit from popover.

https://github.com/ohcnetwork/care_fe/pull/12443/commits/dddca90334e375bb43c7f21336985222ce609b62

yash-learner committed:
Merge branch 'labs-v2' into some-fixes-and-cleanup-in-sdf

yash-learner committed:
Merge branch 'labs-v2' into some-fixes-and-cleanup-in-sdf

yash-learner committed:
Merge branch 'labs-v2' into some-fixes-and-cleanup-in-sdf

abhimanyurajeesh reviewed (changes requested):
https://github.com/user-attachments/assets/7eeb85c4-916e-45be-9d06-006c2abc266e

If the user selects the unit before value, not able to edit the value nor unit.

yash-learner committed:
Merge branch 'labs-v2' into some-fixes-and-cleanup-in-sdf

yash-learner committed:
Fix NaN getting filled in

yash-learner committed:
Merge branch 'labs-v2' into some-fixes-and-cleanup-in-sdf

abhimanyurajeesh reviewed (commented):
LGTM @nihal467

yash-learner committed:
Merge branch 'labs-v2' into some-fixes-and-cleanup-in-sdf

yash-learner commented:
> @yash-learner Why was this merged without testing? cc @nihal467

@amjithtitus09 

I was not aware and didn't notice it, I merged it as we are not doing any testing by @nihal467 while merging into Labs.

From now on will take care of it.
